File: src/Tiantong.Wms.Api/Entities/PurchaseOrder.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Tiantong.Wms.Api
{
  [Table("purchase_orders")]
  public class PurchaseOrder : Entity
  {
    public virtual int warehouse_id { get; set; }

    public virtual string number { get; set; }

    public virtual int operator_id { get; set; }

    public virtual int applicant_id { get; set; }

    public virtual int department_id { get; set; }

    public virtual int supplier_id { get; set; }

    public virtual string status { get; set; } = "";

    public virtual string comment { get; set; } = "";

    public virtual DateTime due_time { get; set; } = DateTime.MinValue;

    public virtual DateTime created_at { get; set; } = DateTime.Now;

    public virtual DateTime finished_at { get; set; } = DateTime.MinValue;

    [ForeignKey("order_id")]
    public virtual List<PurchasePayment> payments { get; set; }

    [ForeignKey("order_id")]
    public virtual List<PurchaseOrderItem> items { get; set; }
  }
}

File: src/Tiantong.Wms.Client/src/Entities/PurchaseOrder.ts
import PurchaseOrderItem from './PurchaseOrderItem'
import PurchaseOrderPayment from './PurchaseOrderPayment'
import { DateTime } from '@/utils/common'

export default class PurchaseOrder {
  id: number = 0

  warehouse_id: number = 0

  number: string | null = null

  operator_id: number = 0

  applicant_id: number = 0

  department_id: number = 0

  supplier_id: number = 0

  status: string = 'created'

  comment: string = ''

  due_time: string = DateTime.now

  created_at: string = DateTime.minValue

  finished_at: string = DateTime.now

  items: Array<PurchaseOrderItem>

  payments: Array<PurchaseOrderPayment>

  constructor (
    payments: Array<PurchaseOrderPayment> = [],
    items: Array<PurchaseOrderItem> = [],
  ) {
    this.items = items
    this.payments = payments
  }
}
